### PR TITLE
Consider topics when routing in Universal Pipeline

### PR DESCRIPTION
**Pull request by jimmyfagan**
This PR adjusts the UP routing code to only consider receivers of the same topic as the sender. The change to the router was _very_ small, so the bulk of the changeset is the adjustments to the RoutingTests.kt file.

Test Steps:

- Confirm that etor sender can send to etor receiver

  1. Reload updated etor settings with this command `./prime multiple-settings set --input ./settings/staging/0149-etor.yml`
  2. Send a lab order report using this postman collection: 
[Temp Collection.postman_collection_json.txt](https://github.com/CDCgov/prime-reportstream/files/11548178/Temp.Collection.postman_collection_json.txt)
  3. Query the submission history to verify that the report is sent to `simulated-lab`, the topic is properly recorded, and check the sftp folder to look for the resulting HL7V2 file.
- Confirm that etor sender cannot send to full-elr receiver
  1. Update  `./settings/staging/0149-etor.yml` to set `simulated-lab`'s topic to `full-elr`.
  2. Reload updated etor settings with this command `./prime multiple-settings set --input ./settings/staging/0149-etor.yml`
  3. Send a lab order report using this postman collection: 
[Temp Collection.postman_collection_json.txt](https://github.com/CDCgov/prime-reportstream/files/11548178/Temp.Collection.postman_collection_json.txt)
  4. Query the submission history to verify that the report was not delivered and the topic is properly recorded.
- Confirm that full-elr sender can send to full-elr receiver
  1. Update  `./settings/staging/0149-etor.yml` to set `etor-service-sender`'s topic to `full-elr`.
  2. Reload updated etor settings with this command `./prime multiple-settings set --input ./settings/staging/0149-etor.yml`
  3. Send a lab order report using this postman collection: 
[Temp Collection.postman_collection_json.txt](https://github.com/CDCgov/prime-reportstream/files/11548178/Temp.Collection.postman_collection_json.txt)
  4. Query the submission history to verify that the report is sent to `simulated-lab`, the topic is properly recorded, and check the sftp folder to look for the resulting HL7V2 file.
- Confirm that full-elr sender cannot send to etor receiver
  1. Update  `./settings/staging/0149-etor.yml` to set `simulated-lab`'s topic to `etor-ti`.
  2. Reload updated etor settings with this command `./prime multiple-settings set --input ./settings/staging/0149-etor.yml`
  3. Send a lab order report using this postman collection: 
[Temp Collection.postman_collection_json.txt](https://github.com/CDCgov/prime-reportstream/files/11548178/Temp.Collection.postman_collection_json.txt)
  4. Query the submission history to verify that the report was not delivered and the topic is properly recorded.

## Changes
- Changed the lambda on the initial retrieval of receivers during `applyFilters` to only include receivers with matching topic to a new input parameter
- Updated RoutingTests.kt to use a helper function and reduce code duplication
- Added tests to cover the 4 cases listed in the "Test Steps" section above

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [x] Added tests?

## Linked Issues
- Addresses #9495

### CODE REVIEW

File: prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt [hunk 1]
```diff
@@ -292,20 +293,21 @@ class FHIRRouter(
...
-    internal fun applyFilters(bundle: Bundle, report: Report): List<Receiver> {
+    internal fun applyFilters(bundle: Bundle, report: Report, topic: Topic): List<Receiver> {
         val listOfReceivers = mutableListOf<Receiver>()
-        // find all receivers that have the full ELR topic and determine which applies
-        val fullElrReceivers = settings.receivers.filter {
+        // find all receivers that have a matching topic and determine which applies
+        val topicReceivers = settings.receivers.filter {
             it.customerStatus != CustomerStatus.INACTIVE &&
-                it.topic.isUniversalPipeline
+                it.topic.isUniversalPipeline &&
```
[the remark this answers is not in the document]
Comment (jimmyfagan) [reply prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt:306]: I basically just left that in as a sanity check, but probably works better to do a `check` and keep this filter simple. Don't think it warrants it's own function at this point, but that's certainly a possibility for the future.

**Comment by jimmyfagan** [comment]
Addressed the in-line comments with the last two commits. On your other comment: smoke tests are a good idea, I just dread messing with those since I do so infrequently. I'll get on that and let you know when it's ready for another look.

**Comment by jimmyfagan** [comment]
Added some smoke tests, ready for re-review!

**Review by mkalish — APPROVED** [review]
Nicely done!  It's great how simple the change ended up being.  Only thought was beefing the smoketest up slightly but not a blocker.

**Review by arnejduranovic — APPROVED** [review]
Looks good!

**Comment by jimmyfagan** [comment]
Updated org settings in staging, and verified smoke tests on staging still succeed.

**Pull request merged**